cpg-core/src/main/java/de/fraunhofer/aisec/cpg/passes/scopes/LoopScope.kt

```diff
@@ -38,7 +38,26 @@ class LoopScope(loopStatement: Statement) :
     var starts: List<Node> = ArrayList()
 
     /** Statements that constitute the start of the Loop condition evaluation, mostly of size 1 */
-    var conditions: List<Node> = ArrayList()
+    val conditions: List<Node>
```

Comment:
Do you know if/where the other variables in this scope class are populated? Is this done in a pass or are they also empty?

Reply:
Looking at the code in the IDE the field is not used, however there is a function `starts()` that is used to return this list of nodes: [Loop starts](https://github.com/Fraunhofer-AISEC/cpg/blob/2666ba530f940d566e9506ea0f6cab96a9f398df/cpg-core/src/main/java/de/fraunhofer/aisec/cpg/passes/scopes/LoopScope.kt#L58-L91)
maybe we can just use that function as the getter. I can't tell you why there is a function and a field. Maybe something went wrong when we did the Java to Kotlin conversion or I refactored once without deleting the field.

Reply:
ah no MB, this is about the `conditions` field, I will leave the comment above in case its useful later, I will take a quick look myself, however, it seems we can delete starts.

Reply:
So this was once used here: [Connect Continues](https://github.com/Fraunhofer-AISEC/cpg/blob/c4239ae4fda4cdbe103695f1848499f30b7f09d7/cpg-core/src/main/java/de/fraunhofer/aisec/cpg/passes/EvaluationOrderGraphPass.kt#L750-L759) and was later replaced with inplace switching, however as you can see it has a different `condition` for the `ForEachStatement`. Yours is probably better/ more intuitive. If you could also change that function to use your `conditions` it would be nice.

Reply:
I changed it as you suggested, see 533c68955d5f4eadf655e23664070977af9bad61. I also added the `AssertStatement` case from [Connect Continues](https://github.com/Fraunhofer-AISEC/cpg/blob/c4239ae4fda4cdbe103695f1848499f30b7f09d7/cpg-core/src/main/java/de/fraunhofer/aisec/cpg/passes/EvaluationOrderGraphPass.kt#L750-L759) to the getter.